File: 2024/day1/day1.rb
#!/usr/bin/env ruby
l = STDIN.readlines.map { |line| line.split.map(&:to_i) }

# part 1
puts l.transpose.map(&:sort).transpose.map {|p| p.reduce(:-).abs}.sum

# part 2
a, b = l.transpose
puts a.map {|x| x * b.count(x)}.sum

File: 2024/day1/day1.py
#!/usr/bin/env python
import sys

a, b = zip(*[list(map(int, line.strip().split())) for line in sys.stdin])

# part 1
print(sum([abs(x - y) for x, y in zip(sorted(a), sorted(b))]))

# part 2
print(sum([x * b.count(x) for x in a]))
